roles/sap_ha_cluster/tasks/main.yml

```diff
@@ -0,0 +1,243 @@
+---
+# 1. Role arguments are validated through meta/arguments_spec.yml (ansible >= 2.11)
+# 2. Detect SAP solution to be configured (scale-up, scale-out, etc.)
+#    * Play hosts must have SAP installed and landscape information discoverable ?!
+# 3. Include tasks for parameter validation for the selected solution
+#    * Certain parameters must match the target solution
+#      (networking, IDs, number of nodes)
+# 4. Include tasks for building the ha_cluster arguments for the selected solution
+#    * scale-up, scale-out, scale-up + mtr, scale-out + mtr
+# 5. Include ha_cluster role with the constructed parameters as role arguments
+# 6. Add solution specific config
+#    * srhook
+# 7. Save 'ha_cluster' input parameters in .yml file for easy re-use
+
+# TODO: Build all resource/constraint configuration variables based on
+# simpler user input (think: drop-down options in a UI)
+# - Make resources configuration modular, allowing additional VIPs, for example
+
+# TODO: create CIB backup before running ha_cluster
+
+# TODO: add parameter to just construct ha_cluster input variables and display the output
+# use-case: users can extract the parameters to add them to their variable files
+# and re-use ha_cluster with a complete set of configuration definitions
+
+# The SAP ID must follow a strict format and not use reserved special values
+# TODO: This check may be better placed in a SAP role earlier in the chain...
```

Comment:
I agree. I will put it into sap_hana_install, see https://github.com/sap-linuxlab/community.sap_install/issues/212 .